deployment/common/changeset/example/link_transfer.go

```diff
@@ -0,0 +1,251 @@
+func transferOrBuildTx(
+	e deployment.Environment,
+	linkState *changeset.LinkTokenState,
+	transfer TransferConfig,
+	opts *bind.TransactOpts,
+	chain deployment.Chain,
+	mcmsConfig *MCMSConfig) (*ethTypes.Transaction, error) {
+	tx, err := linkState.LinkToken.Transfer(opts, transfer.To, transfer.Value)
+	if err != nil {
+		return nil, fmt.Errorf("error packing transfer tx data: %w", err)
+	}
+	// only wait for tx if we are not using MCMS
+	if mcmsConfig == nil {
+		if _, err := deployment.ConfirmIfNoError(chain, tx, err); err != nil {
+			e.Logger.Errorw("Failed to confirm transfer tx", "chain", chain.String(), "err", err)
+			return nil, err
+		}
+	}
+	return tx, nil
+
+}
+
+// LinkTransfer takes the given link transfers and executes them or creates an MCMS proposal for them.
+func LinkTransfer(e deployment.Environment, cfg *LinkTransferConfig) (deployment.ChangesetOutput, error) {
+
+	err := cfg.Validate(e)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("invalid LinkTransferConfig: %w", err)
+	}
+	chainSelectors := []uint64{}
+	for chainSelector := range cfg.Transfers {
+		chainSelectors = append(chainSelectors, chainSelector)
+	}
+	mcmsPerChain := map[uint64]*owner_helpers.ManyChainMultiSig{}
+
+	timelockAddresses := map[mcms.ChainIdentifier]common.Address{}
+	// Initialize state for each chain
+	linkStatePerChain, mcmsStatePerChain, err := initStatePerChain(cfg, e)
+
+	allBatches := []timelock.BatchChainOperation{}
+	for chainSelector := range cfg.Transfers {
+		chainID := mcms.ChainIdentifier(chainSelector)
+		chain := e.Chains[chainSelector]
+		linkAddress := linkStatePerChain[chainSelector].LinkToken.Address()
+		mcmsState := mcmsStatePerChain[chainSelector]
+		linkState := linkStatePerChain[chainSelector]
+
+		timelockAddress := mcmsState.Timelock.Address()
+
+		mcmsPerChain[uint64(chainID)] = mcmsState.ProposerMcm
+
+		timelockAddresses[chainID] = timelockAddress
+		batch := timelock.BatchChainOperation{
+			ChainIdentifier: chainID,
+			Batch:           []mcms.Operation{},
+		}
+
+		opts := getDeployer(e, chainSelector, cfg.McmsConfig)
+		totalAmount := big.NewInt(0)
+		for _, transfer := range cfg.Transfers[chainSelector] {
+			tx, err := transferOrBuildTx(e, linkState, transfer, opts, chain, cfg.McmsConfig)
+			if err != nil {
+				return deployment.ChangesetOutput{}, err
+			}
+			op := mcms.Operation{
+				To:           linkAddress,
+				Data:         tx.Data(),
+				Value:        big.NewInt(0),
+				ContractType: string(types.LinkToken),
+			}
+			batch.Batch = append(batch.Batch, op)
+			totalAmount.Add(totalAmount, transfer.Value)
+		}
+
+		allBatches = append(allBatches, batch)
+	}
+	chainMetadata, err := proposalutils.BuildProposalMetadata(chainSelectors, mcmsPerChain)
+	if err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+	if cfg.McmsConfig != nil {
+		proposal, err := timelock.NewMCMSWithTimelockProposal(
```

Comment:
avoiding refactoring BuildBatches to accept the ValidUntil? I think preferable to just still use that helper, keep build metadata private and just use the default of 72hrs

Reply:
Added here https://github.com/smartcontractkit/chainlink/pull/15512/commits/7c2ecf0d39347a53e68447f5239aa4c0510b9e23